### DISCUSSION TITLE
How to pretty print a dictionary inside a log

### DISCUSSION THREAD

**Discussion by aahnik** [Textualize/rich#1283] [Q&A]
I want to beautifully print a dictionary. I want to get the output of 

```python
from rich.pretty import pprint
pprint(my_dict)
```

But I want to log it as info, not a direct print.

```python
logging.info(my_dict)
```

I have used rich's handler for logging. And logging works perfectly fine. The logs are colorized and styled as I need. But the dictionary is not pretty printed. It is colorized when I log it, but I need the output of pretty print, but inside a log.

How can I do so ?

I tried this, but did not work

```python
from rich.pretty import Pretty
logging.info(Pretty(my_dict))
```

**Comment by willmcgugan** [comment]
You can [capture](https://rich.readthedocs.io/en/latest/console.html#capturing-output) output from Rich and send that to logging. But bear in mind most loggers will not be aware of color codes, and some may break Rich formatting.

**Comment by deadnews** [comment]
```python
from rich.pretty import pretty_repr

logging.info(pretty_repr(my_dict))
````
Alternatively:
```python
from pprint import pformat

logging.info(pformat(my_dict))
````
